docs/tutorials/introduction.rst

```diff
@@ -0,0 +1,307 @@
+we want it to build, and how to do it.
+
+
+.. code-block:: meson
+   :caption: meson.build
+
+   project('purelib-and-platlib', 'c')
+
+   py_mod = import('python')
+   py = py_mod.find_installation()
+
+   py.extension_module(
+       'our_first_module', 'our_first_module.c',
+       install: true,
+   )
+
+
+Here, we use the `Meson Python module`_ to build our ``our_first_module``
+module. We make sure to install it, by passing ``install: true`` to
+``extension_module``, as ``meson-python`` will only include in the binary
+distribution artifacts targets that Meson would install onto system. Having non
+installed targets allows you to build targets for use within the build, or for
+tests.
+
+
+Define the Python build system
+==============================
+
+Now, we need to tell Python packaging tooling what build backend to use to build
+our package. We do this by creating a ``build-system`` section in the
+``pyproject.toml`` file, which is the file used to configure Python packaging
+tooling.
+
+Inside the ``build-system`` section, we need to define two keys,
+``build-backend`` and ``requires``. ``build-backend`` defines which build
+backend should be used for the project, in meson-python it should always have
+the value of ``mesonpy``. ``requires`` lets us specify which packages need to be
+installed for the build process, it should include ``meson-python`` and any
+other dependencies you might need (eg. ``Cython``), in our case it's just
+``meson-python``.
+
+
+.. code-block:: toml
+   :caption: pyproject.toml
+
+   [build-system]
+   build-backend = 'mesonpy'
+   requires = ['meson-python']
+
+
+After we specify which backend to use, we'll want to define the package
+metadata. This is done in the ``project`` section, and the format is pretty
+self-explanatory by looking at our example.
+
+
+.. code-block:: toml
+   :caption: pyproject.toml
+
+   ...
+
+   [project]
+   name = 'our-first-project'
+   version = '0.0.1'
+   description = 'Our first Python project, using meson-python!'
+   readme = 'README.md'
+   requires-python = '>=3.8'
+   license = {file = 'LICENSE.txt'}
+   authors = [
+     {name = 'Bowsette Koopa', email = 'bowsette@example.com'},
+   ]
+
+
+.. admonition:: Adding dependencies
+   :class: seealso
+
+   If you need to add dependencies to the project metadata, you can check our
+   :ref:`how-to-guides-add-dependencies` guide.
+
+
+.. admonition:: Declaring project metadata
+   :class: seealso
+
+   Our example doesn't make use of all the fields available, so we recommend you
+   check out the `PyPA documentation on project metadata`_.
+
+
+.. admonition:: Writing TOML
+   :class: seealso
+
+   If you are not familiar with the TOML configuration language or need some
+   help, be sure to check out
+   `their documentation <https://toml.io/en/v1.0.0>`__.
+
+
+Generating the distribution artifacts
+=====================================
+
+Now that we have a valid Python project, we can release it.
```

Comment:
There is one issue here I think (also related to the FIXME 3 paragraphs down): you go straight from metadata to a release, without local use first. How about inserting a section _"Building and installing our package"_ first? Touching on `pip install .`, mentioning that builds a wheel for the current Python interpreter and then installs it. Show that it's installed with `pip list`, then import it. Conclude that we now have a working package. Next step: a release.